lib/posix/pthread.c

```diff
@@ -274,7 +272,6 @@ int pthread_cancel(pthread_t pthread)
 		} else {
 			thread->retval = PTHREAD_CANCELED;
 			thread->state = PTHREAD_EXITED;
-			pthread_cond_broadcast(&thread->state_cond);
```

Comment:
Why was this condition variable even present? Was there some state transition that couldn't be monitored from the OS thread object itself?

Reply:
The original implementation of pthreads were almost completely orthogonal to k_thread.